src/main/scala/org/constellation/primitives/Transaction.scala

```diff
@@ -75,6 +75,10 @@ case class Transaction(
   val ordinal: Long = lastTxRef.ordinal + 1L
 }
 
+object Transaction {
+  def oFeeToValue(fee: Option[Long]): Long = fee.map(v => if (v < 0L) 0L else v).getOrElse(0L)
```

Comment:
it would be cleaner when you create a function on Transaction class, like:
```
def feeValue: Long = ...
```
so then you're able to use it like:
```
t.amount - t.feeValue
```

Hardcoding type into variable/fn name is kinda ugly.

Reply:
True and I guess it's transient so it doesn't change the schema.

Reply:
`def` is a function, not a property, so it doesn't change the schema by definition.

Reply:
Ah right. I guess in our example `val ordinal: Long = lastTxRef.ordinal + 1L` can be perceived as transient. I miss-understood kryo docs some time ago. Thinking about def's as transient fields that may be serialized...

Reply:
I don't think so. `ordinal` is a property of transaction, evaluated once at the beginning